packages/core/lib/patchers/aws_p.js

```diff
@@ -73,22 +73,29 @@ function captureAWSRequest(req) {
     return req;
   }
 
-  var throttledError = this.throttledError || throttledErrorDefault;
+  const throttledError = this.throttledError || throttledErrorDefault;
 
-  var stack = (new Error()).stack;
-  var subsegment = parent.addNewSubsegment(this.serviceIdentifier);
-  var traceId = parent.segment ? parent.segment.trace_id : parent.trace_id;
+  const stack = (new Error()).stack;
 
-  var buildListener = function(req) {
+  let subsegment;
+  if (parent.notTraced === false || !parent.subsegments[parent.subsegments.length - 1].notTraced) {
```

Comment:
Same here. Can we just look at the subsegment and not have the segment check?

Reply:
See above